pkg/controller.v2/controller_cleanup.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package controller provides a Kubernetes controller for a TFJob resource.
+package controller
+
+import (
+	"time"
+
+	tfv1alpha2 "github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1alpha2"
+	log "github.com/sirupsen/logrus"
+)
+
+// cleanupBot cleans up the jobs that specifies a cleanup timeout after finished.
+func (tc *TFJobController) cleanupBot(stopCh <-chan struct{}) {
+	for {
+		select {
+		case ci := <-tc.cleanupInfoCh:
+			go func() {
```

Comment:
How about check in reconcileTFJobs that  if currentTime-Job.Status.CompletionTIme > timeout,  if true, delete the job. Then we don't need to start a go routine for each timeout.

Reply:
It will be better. If we use goroutine, we can not recover from the crash.